CRM/Event/Form/Registration.php

```diff
@@ -1648,8 +1648,9 @@ public function processFirstParticipant($participantID) {
   protected function checkValidEvent(): void {
     // is the event active (enabled)?
     if (!$this->_values['event']['is_active']) {
-      // form is inactive, die a fatal death
-      CRM_Core_Error::statusBounce(ts('The event you requested is currently unavailable (contact the site administrator for assistance).'));
+      // Form is inactive, redirect to the list of events
+      $urlList = CRM_Utils_System::url('civicrm/event/list', FALSE, NULL, FALSE, TRUE);
+      CRM_Core_Error::statusBounce(ts('The event you requested is currently unavailable (contact the site administrator for assistance).'), $urlList);
```

Comment:
@mlutfy I'm not completely sure this is the right thing to do. It's going to be triggered because the event is disabled but where to redirect seems like it might depend on how you got to the event in the first place (eg. a link in an email, a bot that made up the link etc) and if we force you to a list of events that might not be the right thing.
Happy to be persuaded though :-)

Reply:
Mostly just worried that this might break desired functionality - eg. if you actually want someone to bounce to front page of website for example.

Reply:
My main concern with being redirected to the front page of a website, is that we do not see any error (because the error is not set in the CMS session). So the user has the impression that there was a bug, or even me, as a dev, who was debugging a request from a client, and I had to debug the code to figure it out.

As a user, let's say you wanted to subscribe to a workshop, and it was an old link, then I think being redirected to the list of workshops is the best option.

but I agree this is somewhat opinionated. Some people will say they have a fancier event list, and they don't want the core one. They can do an extension for "redirect the event list".

For other small non-profits who do not have anything fancy, and might be using Standalone, I think the event list is a good option. Eventually, I would like to improve it too. This is a very recurrent request and is very fiddly and often CMS-specific (list events by date and by calendar).

Reply:
Well you've sold it to me. It can always be changed again later!